### PR TITLE
MOEN-10991 Flutter SDK update to 2.0

### PR DESCRIPTION
**Pull request by arshiya-moengage**

### CODE REVIEW

**Review by umangmoe — CHANGES_REQUESTED** [review]

File: core/lib/moengage_flutter.dart [hunk 1]
```diff
@@ -123,7 +124,7 @@ class MoEngageFlutter {
   }
 
   /// Tracks an event with the given attributes.
-  void trackEvent(String eventName, MoEProperties eventAttributes) {
+  void trackEvent(String eventName, MoEProperties? eventAttributes) {
```
[the remark this answers is not in the document]
Comment (arshiya-moengage) [reply core/lib/moengage_flutter.dart:127]: fixed

File: core/lib/moe_android_core.dart [hunk 2]
```diff
@@ -12,15 +12,13 @@ import 'package:moengage_flutter/moe_push_service.dart';
...
-    _channel = channel;
-  }
+  MoEAndroidCore(this._channel);
 
   void enableSDKLogs() {
     _channel.invokeMethod(methodEnableSDKLogs);
   }
 
-  void trackEvent(String eventName, MoEProperties eventAttributes) {
+  void trackEvent(String eventName, MoEProperties? eventAttributes) {
```
[the remark this answers is not in the document]
Comment (arshiya-moengage) [reply core/lib/moe_android_core.dart:21]: fixed

File: core/lib/moengage_flutter.dart [hunk 3]
```diff
@@ -67,45 +68,45 @@ class MoEngageFlutter {
           _onPushTokenGenerated != null) {
         PushToken pushToken = pushTokenFromJson(call.arguments);
         if (pushToken != null) {
-          _onPushTokenGenerated(pushToken);
+          _onPushTokenGenerated!(pushToken);
```
[the remark this answers is not in the document]
Comment (arshiya-moengage) [reply core/lib/moengage_flutter.dart]: fixed

File: core/lib/utils.dart [hunk 4]
```diff
@@ -11,7 +11,7 @@ import 'package:moengage_flutter/inapp_campaign.dart';
 import 'package:moengage_flutter/moe_push_service.dart';
 
 Map<String, dynamic> getEventPayload(
-    String eventName, MoEProperties eventAttributes) {
+    String eventName, MoEProperties? eventAttributes) {
```
[the remark this answers is not in the document]
Comment (arshiya-moengage) [reply core/lib/utils.dart:14]: fixed

File: core/lib/moengage_flutter.dart [hunk 5]
```diff
@@ -67,45 +68,45 @@ class MoEngageFlutter {
           _onPushTokenGenerated != null) {
         PushToken pushToken = pushTokenFromJson(call.arguments);
         if (pushToken != null) {
```
[the remark this answers is not in the document]
Comment (arshiya-moengage) [reply core/lib/moengage_flutter.dart:70]: fixed

File: core/lib/moe_push_service.dart [hunk 6]
```diff
@@ -26,6 +25,7 @@ extension MoEPushServiceExt on MoEPushService {
         return MoEPushService.push_kit;
       case _pushServiceMiPush:
         return MoEPushService.mi_push;
+      default: throw Exception("error: MoEPushService.fromString() : $pushService is not a valid pushService type.");
```
[the remark this answers is not in the document]
Comment (arshiya-moengage) [reply core/lib/moe_push_service.dart:28]: Done

File: inbox/lib/payload_transformer.dart [hunk 7]
```diff
@@ -128,22 +128,21 @@ Map<String, String> mapFromMedia(Media media) {
...
 }
 
-Map<String, dynamic> actionToMap(Action action) {
+Map<String, dynamic>? actionToMap(Action action) {
   switch (action.actionType) {
     case ActionType.navigation:
-      return navigationActionToMap(action);
+      return navigationActionToMap(action as NavigationAction);
   }
-  return null;
```
Comment (ChengappaMoEngage) [inline inbox/lib/payload_transformer.dart:146]: Here Return Null should be present, right?
If the switch case is not satisfied, we should return null here

File: inbox/lib/payload_transformer.dart [hunk 8]
```diff
@@ -78,31 +79,30 @@ TextContent textContentFromMap(Map<String, dynamic> textMap) {
...
 }
 
-Action actionFromMap(Map<String, dynamic> actionMap) {
+Action? actionFromMap(Map<String, dynamic> actionMap) {
   ActionType actionType = ActionTypeExt.fromString(actionMap[ACTION_TYPE]);
   switch (actionType) {
     case ActionType.navigation:
       return navigationActionFromMap(actionType, actionMap);
   }
-  return null;
```
Comment (ChengappaMoEngage) [inline inbox/lib/payload_transformer.dart:105]: Same here return null should be present.
If the switch case is not satisfied, we should return null here

Comment (arshiya-moengage) [reply inbox/lib/payload_transformer.dart:105]: Fixed.
Had removed it by mistake, Flutter code analysis had marked this line as dead code.

File: inbox/lib/payload_transformer.dart (hunk 7 above)
Comment (arshiya-moengage) [reply inbox/lib/payload_transformer.dart:146]: Fixed.
Had removed it by mistake, Flutter code analysis had marked this line as dead code.

**Review by ChengappaMoEngage — APPROVED** [review]
LGTM

**Review by umangmoe — APPROVED** [review]

**Pull request merged**